File: not-designers-main/dataspace-ce/modules/dataspace-legacy/dataspace-core/dataspace-graphql-schema/src/main/java/sbp/com/sbt/dataspace/graphqlschema/Float4ScalarTypeCoercing.java
package sbp.com.sbt.dataspace.graphqlschema;

import graphql.language.FloatValue;
import graphql.language.IntValue;
import graphql.language.StringValue;
import graphql.schema.Coercing;

import java.math.BigDecimal;

import static sbp.com.sbt.dataspace.graphqlschema.Helper.getCoercingParseLiteralException;
import static sbp.com.sbt.dataspace.graphqlschema.Helper.getCoercingParseValueException;

/**
 * Handler for scalar type Float4
 */
class Float4ScalarTypeCoercing implements Coercing<Float, Float> {

  @Override
  public Float serialize(Object dataFetcherResult) {
    if (dataFetcherResult instanceof String) {
      return Float.valueOf((String) dataFetcherResult);
    } else {
      return (Float) dataFetcherResult;
    }
  }

  @Override
  public Float parseValue(Object input) {
    if (input instanceof Number || input instanceof String) {
      try {
        return new BigDecimal(input.toString()).floatValue();
      } catch (Exception e) {
          // None action
      }
    }
    throw getCoercingParseValueException(GraphQLSchemaHelper.FLOAT4_SCALAR_TYPE_NAME, input);
  }

  @Override
  public Float parseLiteral(Object input) {
    if (input instanceof FloatValue) {
      return ((FloatValue) input).getValue().floatValue();
    } else if (input instanceof IntValue) {
      return ((IntValue) input).getValue().floatValue();
    } else if (input instanceof StringValue) {
      try {
        return Float.valueOf(((StringValue) input).getValue());
      } catch (Exception e) {
          // None action
      }
    }
    throw getCoercingParseLiteralException(GraphQLSchemaHelper.FLOAT4_SCALAR_TYPE_NAME, input);
  }
}

File: not-designers-main/dataspace-ce/modules/dataspace-graphql/src/main/kotlin/ru/sbertech/dataspace/graphql/schema/utils/coercing/Float4ScalarTypeCoercing.kt
package ru.sbertech.dataspace.graphql.schema.utils.coercing

import graphql.GraphQLContext
import graphql.execution.CoercedVariables
import graphql.language.FloatValue
import graphql.language.IntValue
import graphql.language.StringValue
import graphql.language.Value
import ru.sbertech.dataspace.graphql.schema.utils.ScalarTypes.Companion.FLOAT4_SCALAR_TYPE_NAME
import java.math.BigDecimal
import java.util.Locale

object Float4ScalarTypeCoercing : BaseCoercing<Float, Float>() {
    override fun serialize(
        dataFetcherResult: Any,
        graphQLContext: GraphQLContext,
        locale: Locale,
    ): Float =
        when (dataFetcherResult) {
            is String -> dataFetcherResult.toFloat()
            else -> (dataFetcherResult as Float).toFloat()
        }

    override fun parseValue(
        input: Any,
        graphQLContext: GraphQLContext,
        locale: Locale,
    ): Float {
        when (input) {
            is Number, is String -> {
                try {
                    return BigDecimal(input.toString()).toFloat()
                } catch (e: Exception) {
                    // do nothing
                }
            }
        }
        throw getCoercingParseValueException(FLOAT4_SCALAR_TYPE_NAME, input)
    }

    override fun parseLiteral(
        input: Value<*>,
        variables: CoercedVariables,
        graphQLContext: GraphQLContext,
        locale: Locale,
    ): Float {
        when (input) {
            is FloatValue -> {
                return input.value.toFloat()
            }

            is IntValue -> {
                return input.value.toFloat()
            }

            is StringValue -> {
                try {
                    return input.value.toFloat()
                } catch (e: Exception) {
                    // do nothing
                }
            }
        }
        throw getCoercingParseLiteralException(FLOAT4_SCALAR_TYPE_NAME, input)
    }
}
